cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -382,41 +389,188 @@ private void computeIncomingSummaryHelper(EndPointField destination, Operation c
+    if (currentOperation.getType() == OperationType.TRANSFORM) {
+      TransformOperation transform = (TransformOperation) currentOperation;
+      for (InputField field : transform.getInputs()) {
+        getIncomingOperationsForFieldHelper(operationsMap.get(field.getOrigin()), visitedOperations);
+      }
+    }
+  }
+
+  /**
+   * <p>Get the subset of operations that used the specified field</p>
+   *
+   * <p>For example if the operation are as follow</p>
+   * <pre>
+   * pRead: personFile -> (offset, body)
+   * parse: body -> (id, name, address)
+   * cRead: codeFile -> id
+   * codeGen: (parse.id, cRead.id) -> id
+   * sWrite: (codeGen.id, parse.name, parse.address) -> secureStore
+   * iWrite: (parse.id, parse.name, parse.address) -> insecureStore
+   * </pre>
+   *
+   * <p>The sourceField is 'id' of codeFile then the returned operations set will contain cRead, codeGen, sWrite.</p>
+   * <p>If the sourceField is 'body' of personFile then the returned set of operations will contain pRead, parse,
+   * codeGen, sWrite, iWrite.</p>
+   *
+   * @param sourceField the {@link EndPointField} whose outgoing operations needs to be found
+   * @return {@link Set} of {@link Operation} which are outgoing from the given sourceField
+   */
+  Set<Operation> getOutgoingOperationsForField(EndPointField sourceField) {
+    if (readOperations == null) {
+      computeAndValidateFieldLineageInfo(this.operations);
+    }
+
+    Set<Operation> visitedOperations = new HashSet<>();
+    for (ReadOperation readOperation : readOperations) {
+      if (readOperation.getSource().equals(sourceField.getEndPoint())) {
+        // if the read operation is for the dataset to which the sourceField belong
+        List<String> outputs = readOperation.getOutputs();
+        for (String output : outputs) {
+          // if the current read operation did read the sourceField for which outgoing operation is requested we
+          // consider it else not.
+          if (output.equals(sourceField.getField())) {
+            visitedOperations.add(readOperation);
+            for (Operation outgoingOperation : operationOutgoingConnections.get(readOperation.getName())) {
+              // check if the outgoing operation does use the sourceField else that even though is an outgoing
+              // operation for the current operation (which reads multiple fields) is not an outgoing operation for
+              // this field as it does not uses it. For example in the above example included in the java doc of this
+              // method the 'parse' transform is an outgoing operation for the 'pread' operation and is also an
+              // outgoing operation for the 'body' field of 'personFile' dataset read by 'pRead' operation but
+              // 'parse' is not an outgoing operation for the field 'offset' of 'personFile' as it does not takes it
+              // as input.
+              if (isUsed(readOperation, outgoingOperation, sourceField.getField())) {
```

Comment:
I think it would be more clear to do something like:

```
InputField inputToOutgoingOperation = new InputField(sourceField.getEndPoint().getName(), sourceField.getField());
if (containsInputField(outgoingOperation, inputToOutgoingOperation)) {
  computeOutgoing(...);
}
```

By the way, it seems like this class ignores the namespace of the endpoint everywhere and just uses the name. Isn't that problematic?

Reply:
I think the class names are little confusing to keep track in head while reading the code. Namespace is only contained in EndPoint which is also a member of EndPointField. We don't do .getName() on EndPoint in this class. 
The confusion might be because of the .getName() of InputField which the name returns the field name and the origin is the Operation name which was the origin of this input field